src/apis/conversations/conversations.controller.ts

```diff
@@ -1,7 +1,51 @@
-import { Controller } from '@nestjs/common';
+import { Body, Controller, Delete, Get, Param, Post, Put, Query, Res, UseGuards } from '@nestjs/common';
 import { ConversationsService } from './conversations.service';
+import { ApiTags } from '@nestjs/swagger';
+import { ConversationDTO } from './dto/conversations.dto';
+import { RoleEntity } from '../role/entity/role.entity';
+import { ConversationEntity } from './entity/conversations.entity';
+import { DeleteConversationDTO } from './dto/delete.conversations.dto';
 
 @Controller('conversations')
+@ApiTags('conversations')
 export class ConversationsController {
-  constructor(private conversationService: ConversationsService) {}
+  constructor(private conversationService: ConversationsService) { }
+
+
+  @Post()
+  async createConversation(@Body() createConversation: ConversationDTO, roleEntity: RoleEntity) {
+    return await this.conversationService.createConversation(createConversation, roleEntity);
+  }
+
+
+  @Get('/search')
+  async searchByName(@Query('name') search: string): Promise<ConversationEntity[]> {
+    return await this.conversationService.findConversationsByName(search);
+  }
+
+
+  @Get()
+  async findAllConversation() {
+    return await this.conversationService.findAll();
+  }
+
+
+  @Get(':userId')
+  async findUserByUserId(@Param('userId') userId: string): Promise<ConversationEntity> {
+    return await this.conversationService.findUserByUserId(userId);
+  }
```

Comment:
why need api :userId

Reply:
I think user login in system will get conversation for that user